src/v2/style-guide/index.md

```diff
@@ -0,0 +1,1805 @@
+  name: 'Todo',
+  // ...
+}
+```
+{% raw %}</div>{% endraw %}
+
+{% raw %}<div class="style-example example-good">{% endraw %}
+#### Good
+
+``` js
+Vue.component('todo-item', {
+  // ...
+})
+```
+
+``` js
+export default {
+  name: 'TodoItem',
+  // ...
+}
+```
+{% raw %}</div>{% endraw %}
+
+
+
+### Component data <sup data-p="a">essential</sup>
+
+**Component `data` must be a function.**
+
+When using the `data` property on a component (i.e. anywhere except on `new Vue`), the value must be a function that returns an object.
+
+{% raw %}
+<details>
+<summary>
+  <h4>Detailed Explanation</h4>
+</summary>
+{% endraw %}
+
+When the value of `data` is an object, it's shared across all instances of a component. Imagine, for example, a `TodoList` component with this data:
+
+``` js
+data: {
+  listTitle: '',
+  todos: []
+}
+```
+
+We might want to reuse this component, allowing users to maintain multiple lists (e.g. for shopping, wishlists, daily chores, etc). There's a problem though. Since every instance of the component references the same data object, changing the title of one list will also change the title of every other list. The same is true for adding/editing/deleting a todo.
+
+Instead, we want each component instance to only manage its own data. For that to happen,each instance must generate a unique data object. In JavaScript, this can be accomplished by returning the object in a function:
+
+``` js
+data: function () {
+  return {
+    listTitle: '',
+    todos: []
+  }
+}
+```
+{% raw %}</details>{% endraw %}
+
+{% raw %}<div class="style-example example-bad">{% endraw %}
+#### Bad
+
+``` js
+Vue.component('some-comp', {
+  data: {
+    foo: 'bar'
+  }
+})
+```
+
+``` js
+export default {
+  data: {
+    foo: 'bar'
+  }
+}
+```
+{% raw %}</div>{% endraw %}
+
+{% raw %}<div class="style-example example-good">{% endraw %}
+#### Good
+``` js
+Vue.component('some-comp', {
+  data: function () {
+    return {
+      foo: 'bar'
+    }
+  }
+})
+```
+
+``` js
+// In a .vue file
+export default {
+  data () {
+    return {
+      foo: 'bar'
+    }
+  }
+}
+```
+
+``` js
+// It's OK to use an object directly in a root
+// Vue instance, since only a single instance
+// will ever exist.
+new Vue({
+  data: {
+    foo: 'bar'
+  }
+})
+```
+{% raw %}</div>{% endraw %}
+
+
+
+### Prop definitions <sup data-p="a">essential</sup>
+
+**Prop definitions should be as detailed as possible.**
+
+In committed code, prop definitions should always be as detailed as possible, specifying at least type(s).
+
+{% raw %}
+<details>
+<summary>
+  <h4>Detailed Explanation</h4>
+</summary>
+{% endraw %}
+
+Detailed [prop definitions](https://vuejs.org/v2/guide/components.html#Prop-Validation) have two advantages:
+
+- They document the API of the component, so that it's easy to see how the component is meant to be used.
+- In development, Vue will warn you if a component is ever provided incorrectly formatted props, helping you catch potential sources of error.
```

Comment:
provided _with_ incorrectly formatted props?

Reply:
The "with" would only be appropriate if we also used an indirect object without "to". [source](https://english.stackexchange.com/questions/137660/provide-vs-provide-with)